apis/hive/v1/clusterdeployment_types.go

```diff
@@ -381,6 +381,12 @@ type ClusterDeploymentStatus struct {
 	// perform the installation.
 	// +optional
 	Platform *PlatformStatus `json:"platformStatus,omitempty"`
+
+	// ClusterVersionStatus is a wholesale copy of the Status section of the spoke cluster's
+	// `clusterversion version` object. This is not officially supported, and is only populated
+	// on request.
+	// +optional
+	ClusterVersionStatus *configv1.ClusterVersionStatus `json:"clusterVersionStatus,omitempty"`
```

Comment:
In case it's relevant to compare with Hypershift API

https://github.com/openshift/hypershift/blob/main/api/hypershift/v1beta1/hostedcluster_types.go#L1787

I'm not familiar with the motivation to create a struct rather than fetching the whole object.

Reply:
Yup, we definitely considered doing it that way. Ultimately we landed on this approach mainly because it makes it so much easier to carry changes forward: we just have to revendor o/api. That's especially easy for hive as we don't operate under the same long release cadence as OCP itself. The only serious concern was CR space, but we convinced ourselves with live data that we're nowhere near the danger zone in that regard.